src/Features/Core/Portable/Snippets/SnippetContext.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Snippets.SnippetProviders;
+
+namespace Microsoft.CodeAnalysis.Snippets;
+
+/// <summary>
+/// The context presented to a <see cref="ISnippetProvider"/> when providing completions.
+/// </summary>
+internal readonly struct SnippetContext
+{
+    internal SnippetContext(SyntaxContext syntaxContext)
+    {
+        SyntaxContext = syntaxContext ?? throw new ArgumentNullException(nameof(syntaxContext));
+    }
+
+    /// <summary>
+    /// The document that the snippet was invoked within.
+    /// </summary>
+    public Document Document => SyntaxContext.Document;
+
+    /// <summary>
+    /// The caret position when the snippet was triggered.
+    /// </summary>
+    public int Position => SyntaxContext.Position;
+
+    internal SyntaxContext SyntaxContext { get; }
```

Comment:
```suggestion
    public SyntaxContext SyntaxContext { get; }
```

Reply:
➡️ This is intentionally `internal`. If `SnippetContext` becomes public (like `CompletionContext` already is), we aren't ready at this point to make `SyntaxContext` public in its current form.